### DISCUSSION TITLE
What is the benefit in migrating to containerd?

### DISCUSSION THREAD

**Discussion by alex1x** [concourse/concourse#7205] [Help & Support] [answered]
Looking at the docs there is some guidance in [migrating from guardian to containerd](https://concourse-ci.org/concourse-worker.html#transitioning-from-guardian-to-containerd) and some comments that starting from 7.4.0 containerd will be the default. What I am missing is any motivation for the migration, as the docs don't give any incentive for switching to containerd (e.g. faster or less cpu intensive or less memory or more features or similar). Best I could find is an [issue on performance comparison](https://github.com/concourse/concourse/issues/5545) which was closed with a comment that there's no performance difference between the two runtimes. Is that still the "official position"? Is it a worthwhile endeavor to migrate? In what circumstances? 

Also I could not find an explicit confirmation that you can run both runtimes at the same time. Last time I tried this I faced issues, but maybe something else was wrong. So should it be possible to have some workers running containerd and some workers running guardian? 

Happy to enrich the docs with such information if you can answer here.

**Comment by edtan** [comment]
I'm also curious about this, but one benefit I have found is that I no longer have to make my workers forward packets (e.g. `iptables -P FORWARD ACCEPT`).  I hit cloudfoundry/guardian#78 when trying to use Guardian, but after switching to containerd, I no longer had to modify my iptables rules.  (My setup is running the Concourse binary via Nomad.)

**Comment by taylorsilva** [comment, marked as answer]
Great question! Like you said, there was hope of some performance benefits but we ended up getting those from overlays metacopy feature, which is completely separate from your choice of runtime.

A second benefit we're hoping to get is that it helps keep the project healthy. There are lots of people developing with containerd and basically no one developing with garden/guardian. By using containerd maybe it'll be easier for people to contribute to the runtime side of Concourse and troubleshoot as well. There's a larger ecosystem around containerd when compared to garden/guardian.

A third benefit was also seeing what it's like to support another runtime. Long-term goals are to also support k8s and maybe even Nomad as runtimes for concourse.
